src/index.ts

```diff
@@ -1,19 +1,31 @@
 import { createReadObject } from './read-object.js';
 import { createRefIndex } from './resolve-ref.js';
 import { createLooseObjectIndex } from './loose-object-index.js';
-import { createPackedObjectIndex } from './packed-object-index.js';
+import { createPackedObjectIndex, CruftPackMode } from './packed-object-index.js';
 import { createFilesMethods } from './files-list.js';
 import { createCommitMethods } from './commits.js';
 import { createStatMethod } from './stat.js';
 
 export * from './types.js';
 export * from './parse-object.js';
-export async function createGitReader(gitdir: string) {
+
+export interface GitReaderParams {
+    /**
+     * Controls the inclusion of cruft packs in packed objects procession.
+     * @see {@link https://git-scm.com/docs/cruft-packs} for more info about cruft packs.
+     * @default 'include'
+     */
+    cruftPacks?: CruftPackMode;
+}
+
+export async function createGitReader(gitdir: string, params?: GitReaderParams) {
+    const paramsWithDefaults = Object.assign({ cruftPacks: 'include' }, params);
```

Comment:
That's a good practice to have an options normalisation function, which may throws when a wrong value for an option or unknown option is passed.

Reply:
An example of the function format would be a huge help here since I can imagine quite a few ways of doing it.

Reply:
got the example in the chat